common/Resources/ti.internal/extensions/node/process.js

```diff
@@ -253,3 +255,103 @@ Ti.App.addEventListener('uncaughtException', function (event) {
 });
 
 export default process;
+
+// Use a nice predictable class/structure for our timers
+// JS engine should be able to optimize easier
+class Timer {
```

Comment:
This name is a little misleading since it's not really a timer. It's a structure to hold a callback function and its arguments so something like `BoundCallback` would fit better IMHO.

In Node the object is called `Immediate` but since we use it for both `setImmediate` and `nextTick` that might cause confusion (although it's only used internally for `nextTick`).